ocaml/typing/value_rec_check.ml

```diff
@@ -988,6 +992,12 @@ let rec expression : Typedtree.expression -> term_judg =
     | Texp_probe_is_enabled _ -> empty
     | Texp_exclave e -> expression e
     | Texp_src_pos -> empty
+    | Texp_overwrite (exp1, exp2) ->
+      join [
+        expression exp1 << Dereference;
```

Comment:
As done elsewhere, it might be nice to write out why you think this is the right thing to put here.

Reply:
It does not matter very much, since we currently mark `Texp_overwrite` as `Dynamic`, so the analysis will always fail if there is an `overwrite_` in a recursive expression. That being said, I think that this code is correct: We dereference the cell to be overwritten, which requires it to be defined outside of the mutually recursive block (and it would not be sound to overwrite a cell that is not yet constructed). But the overwritten expression itself may be recursive. We do not put a guard on here, since this is done by the tuple/record/constructor that is contained in the overwritten expression.

Reply:
Thanks for the explanation. Please capture this in a comment in the code.